tests/storages_tests/test_storages.py

```diff
@@ -1100,6 +1102,28 @@ def test_get_trial_id_from_study_id_trial_number(storage_mode: str) -> None:
         )
 
 
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_set_trial_state_values_thread_safe(storage_mode: str) -> None:
```

Comment:
IMO, I would not name it as `test_set_trial_state_values_thread_safe` because this test is actually testing the thread-safety of `_pop_waiting_trial_id`. Moreover, what we what is the thread-safety of `_pop_waiting_trial_id`, which is called on specifying n_job > 1. It does not really matter whether `set_trial_state_values` is thread-safe or not. We can may be move this test to `test_study.py`?? What do you think?